src/app/FakeLib/TargetHelper.fs

```diff
@@ -322,36 +324,114 @@ let listTargets() =
 // Instead of the target can be used the list dependencies graph parameter.
 let doesTargetMeansListTargets target = target = "--listTargets"  || target = "-lt"
 
+/// Determines a parallel build order for the given set of targets
+let determineBuildOrder (d : TargetTemplate<unit>) =
```

Comment:
Hi there,
I implemented some very basic tests for the parallel build order checking if it maintains dependencies correctly.
However I did generate the graphs by hand and cleared the TargetDict manually before running the test.
Please let me know if you have any further suggestions.